### ISSUE TITLE
I think reward scaling not working in FinRL env --> early end to training

### ISSUE THREAD

**Issue by asi212** [AI4Finance-LLC/ElegantRL#18]
Hello,
I am using the FinRL environment with a custom dataset (I have 20,000 time periods of training data -- np array shape is [200000, 6*43])

Right now, the training stops after 20,000 time peroids/1 epoch.  I think this is because the MaxR becomes more than TargetR, so the training stops.

I try setting args.reward_scale = 1e-20, a very very low number, but the reward for the epoch does not change. It is the same reward as when args.reward_scale = 1.

I checked, and the args.reward_scale variable is passed to run.py. After that, I have a difficult time understanding the code.

Also, is there a way to manually set the # of training epochs? Or the # of time steps? Thank you

![image](https://user-images.githubusercontent.com/51300707/111056603-f3a1a900-8445-11eb-856d-145a999be686.png)

`from elegantrl.run import Arguments, train_and_evaluate, train_and_evaluate__multiprocessing
from elegantrl.tutorial.env import PreprocessEnv
import gym
gym.logger.set_level(40)  # Block warning

args = Arguments(if_on_policy=True)
'''choose an DRL algorithm'''
from elegantrl.tutorial.agent import AgentPPO
args.agent = AgentPPO()

from elegantrl.env import FinanceMultiStockEnv  # a standard env for ElegantRL, not need PreprocessEnv()
args.env = FinanceMultiStockEnv(stock_dim=6, if_train=True, max_stock=1e2, transaction_fee_percent=1e-3,
                                train_beg=0, train_len=20000, info_col_length=43)
args.env_eval = FinanceMultiStockEnv(stock_dim=6,if_train=False, max_stock=1e2, transaction_fee_percent=1e-3,
                                     train_beg=0, train_len=20000, info_col_length=43)  # eva_len = 1699 - train_len
args.reward_scale = 1e-20 #2 ** 0  # RewardRange: 0 < 1.0 < 1.25 <
args.break_step = int(5e6)
args.max_step = args.env.max_step
args.max_memo = (args.max_step - 1) * 8
args.batch_size = 2 ** 11
"TotalStep:  2e5, TargetReward: 1.25, UsedTime:  200s"

'''train and evaluate'''
train_and_evaluate(args)
# args.rollout_num = 8
#train_and_evaluate__multiprocessing(args)  # try multiprocessing in formal version`

**Comment by Yonv1943** [comment]
Hi,

> I am using the FinRL environment with a custom dataset (I have 20,000 time periods of training data -- np array shape is [200000, 6*43])

- If you train and evaluate your agent using same data. The agent will get a very high episode return. So we set 60% for training, 40% for evaluate. The episode return is 1.0 to 1.6. We set the `target_reward` as 1.25 in default. **You can change it** using `args.env.target_reward = 123456`
- Your np array shape is [200000, 6*43], such a long data will more easy to get high reward, Our demo custom env is 60% (length=1024) for training and 40% (length=675) for evaluating. If we train and evaluate agent on same data, the target reward will be over 20.

> I try setting args.reward_scale = 1e-20, a very very low number, but the reward for the epoch does not change. It is the same reward as when args.reward_scale = 1.

Change `reward_scale` won't affect episode return. We have `self.episode_return = next_total_asset / self.initial_account`. When agent reached `target_reward=1.25`, **it means that it get 1.25 times `initial_account` in evaluated data. **

In addition,  `reward_scale` is just an important hyperparameters for SAC. It hardly affects the PPO because of the reward normalization in PPO `buf_advantage = (buf_advantage - buf_advantage.mean()) / (buf_advantage.std() + 1e-5)`.


PPO don't need to change hyperparameters `reward_scale`.

**Comment by asi212** [comment]
Thank you very much for your detailed answer. I understand now. Now, I have no problem with the training. Thank you.
